[Usage] Mold Usage 서버 DB 에러

<!--
먼저 GitHub에 작성하고자 하는 이슈 및 요청이 이미 등록되어 있는지 확인하십시오.
또한, 최신 릴리즈 및 기본 브랜치에 해당 사항이 반영되어 있는지도 확인하십시오.
이슈를 작성한 후에 발견한 정보는 코멘트에 남겨 주십시오. 작성된 코멘트는 삭제하지 않아도 됩니다.
-->

#### 이슈 타입
<!-- 아래에서 해당하는 타입을 선택하고 나머지는 지우세요. 복수선택 가능 -->
 * 버그 및 오류 보고
 
#### 컴포넌트 이름
<!--
이슈를 카테고리화 하십시오. 예를 들어, API, VR, VM, VPN, UI 등.
-->
~~~
USAGE SERVER
~~~

#### ABLESTACK 버전
<!--
새 라인으로 구분하여 이슈가 영향을 받는 버전, 기본 브랜치 상의 커밋 아이디, 이슈 아이디 등을 기재하십시오.
-->

~~~
4.0
~~~

#### 구성
<!--
이슈와 관련된 구성 정보를 기재하십시오 예, 기본 네트워크, 확장 네트워크, iSCSI, NFS 등.
-->
~~~
Usage Server 구성 및 실행 구성
~~~

#### OS / 환경
<!--
이슈와 관련된 운영체제 및 환경 정보를 기재하십시오. 없다면 N/A를 기재합니다.
-->
~~~
N/A
~~~

#### 이슈 내용
<!-- 문제 또는 기능에 대한 설명을 간단하게 작성합니다. -->

Usage 서버에서 스냅샷 사용량 측정 시 로그에 다음의 에러가 표시됨.

~~~
2024-01-29 12:15:05,021 ERROR [cloud.usage.UsageManagerImpl] (Usage-Job-1:null) (logid:) Exception in usage manager
com.cloud.utils.exception.CloudRuntimeException: DB Exception on: com.mysql.cj.jdbc.ClientPreparedStatement: SELECT snapshots.id, snapshots.data_center_id, snapshots.account_id, snapshots.domain_id, snapshots.volume_id, snapshots.disk_offering_id, snapshots.name, snapshots.status, snapshots.snapshot_type, snapshots.location_type, snapshots.type_description, snapshots.size, snapshots.created, snapshots.removed, snapshots.hypervisor_type, snapshots.version, snapshots.uuid, snapshots.min_iops, snapshots.max_iops FROM snapshots WHERE snapshots.id = 2  AND snapshots.removed IS NULL
        at com.cloud.utils.db.GenericDaoBase.findById(GenericDaoBase.java:1041)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.CGLIB$findById$80(<generated>)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96_FastClassByCloudStack_daa87d54.invoke(<generated>)
        at net.sf.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
        at com.cloud.utils.component.ComponentInstantiationPostProcessor$InterceptorDispatcher.intercept(ComponentInstantiationPostProcessor.java:122)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.findById(<generated>)
        at com.cloud.utils.db.GenericDaoBase.lockRow(GenericDaoBase.java:1018)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.CGLIB$lockRow$48(<generated>)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96_FastClassByCloudStack_daa87d54.invoke(<generated>)
        at net.sf.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
        at com.cloud.utils.component.ComponentInstantiationPostProcessor$InterceptorDispatcher.intercept(ComponentInstantiationPostProcessor.java:122)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.lockRow(<generated>)
        at com.cloud.utils.db.GenericDaoBase.findById(GenericDaoBase.java:964)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.CGLIB$findById$82(<generated>)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96_FastClassByCloudStack_daa87d54.invoke(<generated>)
        at net.sf.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
        at com.cloud.utils.component.ComponentInstantiationPostProcessor$InterceptorDispatcher.intercept(ComponentInstantiationPostProcessor.java:122)
        at com.cloud.storage.dao.SnapshotDaoImpl_EnhancerByCloudStack_aeca7a96.findById(<generated>)
        at com.cloud.usage.UsageManagerImpl.createSnapshotHelperEvent(UsageManagerImpl.java:1694)
        at com.cloud.usage.UsageManagerImpl.createHelperRecord(UsageManagerImpl.java:1052)
        at com.cloud.usage.UsageManagerImpl.parse(UsageManagerImpl.java:706)
        at com.cloud.usage.UsageManagerImpl.runInContextInternal(UsageManagerImpl.java:421)
        at com.cloud.usage.UsageManagerImpl$1.runInContext(UsageManagerImpl.java:363)
        at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:48)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:102)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52)
        at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:45)
        at com.cloud.usage.UsageManagerImpl.run(UsageManagerImpl.java:365)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.sql.SQLSyntaxErrorException: Table 'cloud_usage.snapshots' doesn't exist
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
~~~

#### 재현 과정
<!--
버그/오류 이슈라면, 문제를 어떻게 재현할 수 있는지 정확하게 보여줘야 합니다. 필요하다면 최소한의 테스트 케이스와 스크린 샷을 사용하십시오.

새로운 기능에 대한 이슈라면, 기능이 어떻게 사용되야 하는지를 보여주십시오.
-->

<!-- 예제가 될 수 있는 플레이북 또는 명령행을 작성하십시오. -->
~~~
systemctl status cloudstack-usage 상태 확인 후 
/var/log/cloudstack/usage/usage.log 내용의 Error 확인
~~~

#### 기대한 결과
<!-- 위의 과정을 실행했을 때 기대한 결과가 무엇입니까? -->

~~~
정상 실행
~~~

#### 실제 결과
<!-- 실제로 발생한 결과는 무엇입니까? -->

<!-- 실행한 명령행 등의 결과를 아래에 붙여 넣습니다. -->
~~~
데이터베이스 에러 표시
~~~

--- a/usage/src/main/java/com/cloud/usage/UsageManagerImpl.java
+++ b/usage/src/main/java/com/cloud/usage/UsageManagerImpl.java
@@ -16,9 +16,6 @@
 // under the License.
 package com.cloud.usage;
 
-import com.cloud.storage.Snapshot;
-import com.cloud.storage.SnapshotVO;
-import com.cloud.storage.dao.SnapshotDao;
 import com.cloud.alert.AlertManager;
 import com.cloud.event.EventTypes;
 import com.cloud.event.UsageEventDetailsVO;
@@ -166,8 +163,6 @@ public class UsageManagerImpl extends ManagerBase implements UsageManager, Runna
     private QuotaAlertManager _alertManager;
     @Inject
     private QuotaStatement _quotaStatement;
-    @Inject
-    private SnapshotDao _snapshotDao;
 
     @Inject
     private BucketStatisticsDao _bucketStatisticsDao;

--- a/usage/src/main/java/com/cloud/usage/UsageManagerImpl.java
+++ b/usage/src/main/java/com/cloud/usage/UsageManagerImpl.java
@@ -1691,12 +1691,6 @@ public class UsageManagerImpl extends ManagerBase implements UsageManager, Runna
 
         long snapId = event.getResourceId();
 
-        SnapshotVO snapshotInstance = _snapshotDao.findById(snapId);
-
-        if (snapshotInstance != null && snapshotInstance.getSnapshotType() == Snapshot.Type.INTERNAL.ordinal()) {
-            return;
-        }
-
         if (EventTypes.EVENT_SNAPSHOT_CREATE.equals(event.getType())) {
             if (usageSnapshotSelection){
                 snapSize =  event.getVirtualSize();

